deepmind/mapper.go

```diff
@@ -115,34 +117,42 @@ func mapValidators(srcValidators []*types.Validator) ([]*codec.Validator, error)
 	return validators, nil
 }
 
-func mapValidator(v *types.Validator) (*codec.Validator, error) {
-	nPK := &codec.PublicKey{}
+func mapValidator(v *types.Validator) (*pbcodec.Validator, error) {
+	nPK := &pbcodec.PublicKey{}
 
 	key := v.PubKey
 	switch key.Type() {
 	case ed25519.KeyType:
-		nPK = &codec.PublicKey{
-			Sum: &codec.PublicKey_Ed25519{Ed25519: key.Bytes()}}
+		nPK = &pbcodec.PublicKey{
+			Sum: &pbcodec.PublicKey_Ed25519{Ed25519: key.Bytes()}}
 	case secp256k1.KeyType:
-		nPK = &codec.PublicKey{
-			Sum: &codec.PublicKey_Secp256K1{Secp256K1: key.Bytes()}}
+		nPK = &pbcodec.PublicKey{
+			Sum: &pbcodec.PublicKey_Secp256K1{Secp256K1: key.Bytes()}}
 	default:
 		return nil, fmt.Errorf("given type %T of PubKey mapping doesn't exist ", key)
 	}
 
 	// NOTE: See note in mapValidatorUpdate() about ProposerPriority
 
-	return &codec.Validator{
+	return &pbcodec.Validator{
 		Address:          v.Address,
 		PubKey:           nPK,
 		VotingPower:      v.VotingPower,
 		ProposerPriority: 0,
 	}, nil
 }
 
-func mapTimestamp(time time.Time) *codec.Timestamp {
-	return &codec.Timestamp{
+func mapTimestamp(time time.Time) *pbcodec.Timestamp {
+	return &pbcodec.Timestamp{
 		Seconds: time.Unix(),
 		Nanos:   int32(time.UnixNano() - time.Unix()*1000000000),
 	}
 }
+
+func mapTx(bytes []byte) (*tx.Tx, error) {
```

Comment:
Why not mapping the Tx directly in `encodeTx` ? I don't see this being reused anywhere.

Reply:
Just in case it would be needed elsewhere in the future. Also tried to follow same pattern as the rest of instrumentation.